src/components/Shared/ScatterPlot/ScatterPlotComponent.tsx

```diff
@@ -431,18 +449,127 @@ export class ScatterPlotComponent extends React.Component<ScatterPlotComponentPr
         const mouseMoveDist = {x: Math.abs(mouseEvent.offsetX - this.stageClickStartX), y: Math.abs(mouseEvent.offsetY - this.stageClickStartY)};
         if (mouseMoveDist.x < DRAG_THRESHOLD && mouseMoveDist.y < DRAG_THRESHOLD) {
             this.onStageClick(ev);
-        } 
+        } else {
+            if (this.props.data) {
+                this.stageClickStartX = undefined;
+                this.stageClickStartY = undefined;
+                if (this.isSelecting && this.zoomMode !== ZoomMode.NONE) {
+                    let minCanvasSpace = Math.min(this.selectionBoxStart.x, this.selectionBoxEnd.x);
+                    let maxCanvasSpace = Math.max(this.selectionBoxStart.x, this.selectionBoxEnd.x);
+                    let minX = this.getValueForPixelX(minCanvasSpace);
+                    let maxX = this.getValueForPixelX(maxCanvasSpace);
+
+                    minCanvasSpace = Math.min(this.selectionBoxStart.y, this.selectionBoxEnd.y);
+                    maxCanvasSpace = Math.max(this.selectionBoxStart.y, this.selectionBoxEnd.y);
+                    // Canvas space y-axis is inverted, so min/max are switched when transforming to graph space
+                    let minY = this.getValueForPixelY(maxCanvasSpace);
+                    let maxY = this.getValueForPixelY(minCanvasSpace);
+
+                    if (this.zoomMode === ZoomMode.X) {
+                        this.props.graphZoomedX(minX, maxX);
+                    }
+                    if (this.zoomMode === ZoomMode.Y) {
+                        this.props.graphZoomedY(minY, maxY);
+                    } else if (this.zoomMode === ZoomMode.XY) {
+                        this.props.graphZoomedXY(minX, maxX, minY, maxY);
+                    }
+                }
+            }
+        }
         this.endInteractions();
     };
 
+    onStageWheel = (ev) => {
+        if (this.props.data && this.props.scrollZoom && this.props.graphZoomedXY && this.chartArea) {
+            const wheelEvent: WheelEvent = ev.evt;
+            const chartArea = this.chartArea;
+            const lineHeight = 15;
+            const zoomSpeed = 0.001;
```

Comment:
I think zoom speed should be a user preference. However, we should handle this is a separate PR, and we should make sure it integrates properly with @duidae's changes to the preference handling. 

Can you file an issue for this, so that we can keep track of it? Zoom speed should be adjustable in the user preferences dialog, and should be consistent between the plots and image view